salt/metalk8s/orchestrate/deploy_node.sls

```diff
@@ -113,6 +113,23 @@ Wait minion available:
 
 {%- endif %}
 
+# Prepare etcd and dependencies
+{%- if 'etcd' in pillar.get('metalk8s', {}).get('nodes', {}).get(node_name, {}).get('roles', []) %}
+
+Prepare etcd dependencies:
+  salt.runner:
+    - tgt: {{ node_name }}
+    - sls: metalk8s.kubernetes.etcd.prepared
+    - pillar: {{ pillar | json }}
```

Comment:
That cannot work ! 
The `metalk8s.orchestrate.register_etcd` need to run on the salt-master not on the minion so you cannot use `salt.runner` targetting a node.

IMO you should call the register here and remove all `prepared` stuff as the prepare is only registering and cannot run on the salt-minion